_episodes/18.md

```diff
@@ -0,0 +1,560 @@
+customer.
+
+|custkey|name|nationkey|mktsegment|sumtotalprice|openstatuscount|failedstatuscount|partialstatuscount|
+|-------|----|---------|----------|-------------|---------------|-----------------|------------------|
+|376|Customer#000000376|16|AUTOMOBILE|1600696.4700000002|3|6|1|
+|377|Customer#000000377|23|MACHINERY|803271.9400000001|3|6|0|
+|379|Customer#000000379|7|AUTOMOBILE|3155009.54|7|11|0|
+
+
+From here, there are many ways you could further evaluate the resulting data. 
+You could filter and look at which market segment is spending the most on your
+products. You could also look at where there are the most failed orders by the
+nation column to evaluate where shipping lines may need to be improved. The 
+table above which results from the example query is a good intermediate state of 
+the data that can be reused for many future evaluations. Instead of defining a 
+new table, you can create a view on this data that will encapsulate the complex
+SQL that was used to calculate it. This is done using the `CREATE VIEW` 
+statement.
+
+```
+CREATE VIEW customer_orders_view AS 
+<complex SQL query above>
+```
+
+Now, when you want to run any further analysis on this intermediate dataset, you
+simply refer to the view instead of having to rewrite the statement before. As
+mentioned, this view also has a schema and is treated much like a table when the
+query engine does its planning. In this way it is also easier to map the data to
+the application logic by enabling different shapes of the same data.
+
+Another reason why you would want to create a view is to control access to the
+data. When running the query, you get to choose which columns and rows get
+filtered out and that return from when users query the view. The authorization 
+of a user is only to read the abstraction of the data rather than the real data.
+
+When creating a view, there are two modes that the view can run in that will 
+indicate the user that will run the queries defined in the view during query
+runtime. You can either run this query as the `DEFINER` which indicates to run
+the view query as the user that created the view, or as the `INVOKER`, which
+indicates to run the view query as the user that is running the outer query of
+the view. The default mode is `DEFINER`. See more 
+[here](https://trino.io/docs/current/sql/create-view.html#security).
+
+There are two types of views; materialized and logical views. The view defined
+above is the standard logical view that gets expanded into its definition. 
+Logical views don't provide any performance benefit since the data is not stored
+upfront and instead queried at query time. Materialized views persist the view 
+data upon view creation by storing the query data on disk.
+
+Schemas for logical views can't be quickly referenced as was done for the table 
+using `SHOW CREATE TABLE`. The query engine can quickly derive each of these
+types. The `SHOW CREATE VIEW` only shows the original query that was used to 
```

Comment:
this sentence is confusing..

Reply:
Removing...it doesn't add much and I found out the point I was trying to make isn't true in all cases :)